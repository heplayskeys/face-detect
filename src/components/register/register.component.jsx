import React, { useState } from 'react';
import { connect } from 'react-redux';
import { updateRoute } from '../../redux/route/route.actions';
import './register.styles.scss';

export const Register = ({ URL, route, setRoute }) => {
	const [errorState, setErrorState] = useState('');

	const handleSubmit = async () => {
		const name = document.querySelector('#name').value;
		const email = document.querySelector('#email-address').value;
		const password = document.querySelector('#password').value;
		const confirmPassword = document.querySelector('#confirm-password').value;

		setErrorState('');

		if (
			name === '' ||
			email === '' ||
			password === '' ||
			confirmPassword === ''
		) {
			console.error('Error: Please fill out all fields');
			setErrorState('Please fill out all fields');
			return;
		}

		const resp = await fetch(`${URL}/register`, {
			method: 'POST',
			headers: { 'Content-Type': 'application/json' },
			body: JSON.stringify({
				name,
				email,
				password,
				confirmPassword
			})
		});

		if (resp.status === 200) {
			alert('User created. Please sign in.');
			setRoute(route);
		} else {
			const errorMsg = await resp.json();
			console.error('Error:', errorMsg);
			setErrorState(`${errorMsg}`);
		}
	};

	const handleEnterKey = event => {
		if (event.key === 'Enter') {
			handleSubmit();
		}
	};

	return (
		<div>
			<article
				id='form-container'
				className='mw6 center bg-white-20 br3 pa3 pa4-ns mv3 ba b--black-10 shadow-5'
			>
				<main className='pa4 black-80'>
					<div className='measure'>
						<fieldset id='sign_up' className='ba b--transparent ph0 mh0'>
							<legend className='f2 fw6 ph0 mh0'>Register</legend>
							<div className='mt3'>
								<label className='db fw6 lh-copy f6' htmlFor='name'>
									Name
								</label>
								<input
									className='pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100'
									maxLength={125}
									type='text'
									name='name'
									id='name'
									onKeyPress={handleEnterKey}
								/>
							</div>
							<div className='mt3'>
								<label className='db fw6 lh-copy f6' htmlFor='email-address'>
									Email
								</label>
								<input
									className='pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100'
									type='email'
									name='email-address'
									id='email-address'
									onKeyPress={handleEnterKey}
								/>
							</div>
							<div className='mv3'>
								<label className='db fw6 lh-copy f6' htmlFor='password'>
									Password
								</label>
								<input
									className='b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100'
									type='password'
									name='password'
									id='password'
									onKeyPress={handleEnterKey}
								/>
							</div>
							<div className='mv3'>
								<label className='db fw6 lh-copy f6' htmlFor='confirm-password'>
									Confirm Password
								</label>
								<input
									className='b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100'
									type='password'
									name='confirm-password'
									id='confirm-password'
									onKeyPress={handleEnterKey}
								/>
							</div>
							<div className='mt4'>
								<input
									className='b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib'
									type='submit'
									value='Register'
									onClick={handleSubmit}
								/>
							</div>
							<div className='lh-copy mt3'>
								<p
									className='f5 link black db bolder pointer'
									onClick={() => setRoute(route)}
								>
									{'<-- Back'}
								</p>
							</div>
						</fieldset>
					</div>
				</main>
			</article>
			{errorState ? (
				<div
					id='error-message'
					className='f4 mt4 dark-red animate__animated animate__heartBeat dib'
					style={{ whiteSpace: 'pre-wrap', textAlign: 'left' }}
				>
					{errorState}
				</div>
			) : null}
		</div>
	);
};

const mapStateToProps = state => ({
	route: state.setRoute.route
});

const mapDispatchToProps = dispatch => ({
	setRoute: route => dispatch(updateRoute(route))
});

export default connect(mapStateToProps, mapDispatchToProps)(Register);
