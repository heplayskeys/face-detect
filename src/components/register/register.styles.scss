$breakpoint-phone: 450px;

#form-container {
	@media screen and (max-width: $breakpoint-phone) {
		margin-top: 10px;
		border: none;
		background: none;
		box-shadow: none;
		margin-bottom: 0;
	}
}

#error-message {
	@media screen and (max-width: $breakpoint-phone) {
		margin: 0;
		font-size: 0.85rem;
	}
}
