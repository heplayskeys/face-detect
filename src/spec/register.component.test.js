import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Register } from '../components/register/register.component';

let wrapper;

describe('<Register />', () => {
	beforeAll(() => {
		const mockProps = {
			URL: '',
			route: '',
			setRoute: jest.fn()
		};
		wrapper = shallow(<Register {...mockProps} />);
	});

	it('renders register component', () => {
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});
});
