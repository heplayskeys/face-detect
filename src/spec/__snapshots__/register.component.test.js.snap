// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> renders register component 1`] = `
<div>
  <article
    className="mw6 center bg-white-20 br3 pa3 pa4-ns mv3 ba b--black-10 shadow-5"
    id="form-container"
  >
    <main
      className="pa4 black-80"
    >
      <div
        className="measure"
      >
        <fieldset
          className="ba b--transparent ph0 mh0"
          id="sign_up"
        >
          <legend
            className="f2 fw6 ph0 mh0"
          >
            Register
          </legend>
          <div
            className="mt3"
          >
            <label
              className="db fw6 lh-copy f6"
              htmlFor="name"
            >
              Name
            </label>
            <input
              className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100 hvr-blk"
              id="name"
              maxLength={125}
              name="name"
              onKeyPress={[Function]}
              type="text"
            />
          </div>
          <div
            className="mt3"
          >
            <label
              className="db fw6 lh-copy f6"
              htmlFor="email-address"
            >
              Email
            </label>
            <input
              className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100 hvr-blk"
              id="email-address"
              name="email-address"
              onKeyPress={[Function]}
              type="email"
            />
          </div>
          <div
            className="mv3"
          >
            <label
              className="db fw6 lh-copy f6"
              htmlFor="password"
            >
              Password
            </label>
            <input
              className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100 hvr-blk"
              id="password"
              name="password"
              onKeyPress={[Function]}
              type="password"
            />
          </div>
          <div
            className="mv3"
          >
            <label
              className="db fw6 lh-copy f6"
              htmlFor="confirm-password"
            >
              Confirm Password
            </label>
            <input
              className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100 hvr-blk"
              id="confirm-password"
              name="confirm-password"
              onKeyPress={[Function]}
              type="password"
            />
          </div>
          <div
            className="mt4"
          >
            <input
              className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib"
              onClick={[Function]}
              type="submit"
              value="Register"
            />
          </div>
          <div
            className="lh-copy mt3"
          >
            <p
              className="f5 link black db bolder pointer"
              onClick={[Function]}
            >
              &lt;-- Back
            </p>
          </div>
        </fieldset>
      </div>
    </main>
  </article>
</div>
`;
